### PR TITLE
fix: unable to edit task that have read-only tags

### PR DESCRIPTION
**Pull request by rbeuque74**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Bug fix

* **What is the current behavior?** (You can also link to an open issue here)
When a subtask is spawned, some default tags are assigned to the subtask, and they are only read-only.
We can't edit the task anymore without removing the read-only tag, which is annoying.

* **What is the new behavior (if this is a feature change)?**
Handle correctly the task edition without touching the read-only tags.

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
No

* **Other information**:

### CODE REVIEW

File: api/handler/task.go [hunk 1]
```diff
@@ -270,9 +271,13 @@ func UpdateTask(c *gin.Context, in *updateTaskIn) (*task.Task, error) {
 	t.SetInput(clearInput)
 	t.SetWatcherUsernames(in.WatcherUsernames)
 
-	if err := utils.ValidateTags(in.Tags); err != nil {
-		return nil, err
+	// validate read-only tags
+	v, readOnlyTagUpdated := in.Tags[constants.SubtaskTagParentTaskID]
+	oldValue, readOnlyTagInTask := t.Tags[constants.SubtaskTagParentTaskID]
+	if (readOnlyTagUpdated && (!readOnlyTagInTask || oldValue != v)) || (!readOnlyTagUpdated && readOnlyTagInTask) {
```
Comment (wI2L) [inline api/handler/task.go:277]: Not sure about the second part of the condition:`(!readOnlyTagUpdated && readOnlyTagInTask)`. This implies that during an update, the read-only tag must be given and be the same value. Is it intentional ?

Comment (loopfz) [reply api/handler/task.go:277]: It seems right to me, their absence would mean a deletion otherwise?

**Comment by wI2L** [comment]
Yes, you're right, I was thinking of a PATCH behavior.

Le lun. 21 sept. 2020 à 20:04, Thomas Schaffer <notifications@github.com> a
écrit :

File: api/handler/task.go (hunk 1 above)
Comment (rbeuque74) [reply api/handler/task.go:277]: Exactly :+1:

**Comment by rbeuque74** [comment]
Can I consider that no comments means LGTM ? :D

**Review by wI2L — APPROVED** [review]

**Pull request merged**